src/main.cpp

```diff
@@ -10,20 +10,41 @@ static int node_counter;
 
 int main(int argc, char* argv[])
 {
+    string input = " ";
+    char mode = 0;
+    Evaluator e;
+
+    // test strings
     string infix = "((10 + 20) * 30)";
     string infix2 = "( ( (2 + 2) + (2 + 2) ) * ( (10 / 5) * 3) )";
     string postfix = "10 20 + 30 *";
     string prefix = "+ 10 20";
 
-    // Knoten: 5
-    // Tiefe: 3
+    // check if enough arguments are given
+    if (argc < 3) {
+        std::cout << "Zu wenige Argumente angegeben." << std::endl;
+        return 1;
+    }
 
+    // read the input string and the mode
+    // spacings are allowed in the input string
```

Comment:
ersetzen durch:
// read the inout string and combine all arguments to the expression, even if they are separated by spaces